### ISSUE TITLE
[Discussion] Should we add Vue to laravel-mix dependencies?

### ISSUE THREAD

**Issue by adiachenko** [JeffreyWay/laravel-mix#680]
It's known that `vue` version must match with the one of `vue-loader` and `vue-template-compiler`. Otherwise, it won't work and will lead to warnings (errors) on compile. 

The problem may not be apparent to you right away, but when using yarn or shrinkwrap and doing frequent updates version mismatch errors will inevitably pop up. For example, here is yarn locking down `vue-template-compiler`:

```
vue-template-compiler@^2.0.0:
  version "2.2.4"
  resolved "https://registry.yarnpkg.com/vue-template-compiler/-/vue-template-compiler-2.2.4.tgz#2856fb09f1571e9098872bf3c512d670eeeafba9"
  dependencies:
    de-indent "^1.0.2"
    he "^1.1.0"
```
Having vue dependency in your application's `package.json` kinda assumes that I may update vue from `2.2.4` to `2.2.6`, but that won't actually work, because `vue-template-compiler` is stuck on the old version.

Now,  you can simply install `vue-loader` and `vue-template-compiler` along with `vue` so you can update them together, but that doesn't seems very reliable, because there is no guarantee that Mix supports specific `vue-loader` at specific point in time. 

In short, I am wondering whether there is a way to make the whole setup less error prone. For example, by explicitly adding Vue to Mix dependencies instead of doing [this](https://github.com/laravel/laravel/blob/v5.4.16/package.json#L17)? I understand that this suggestion is somewhat controversial so I'm looking for more opinions.

**Comment by adiachenko** [comment]
Closing this, because I found a kinda obvious and more clean way to solve version mismatch issues with Vue and Mix by doing the following after `yarn upgrade`:

```
yarn remove laravel-mix
yarn add laravel-mix
```
